src/utils/forecast.js

```diff
@@ -1,17 +1,17 @@
 const request = require('request')
 
-const forecast = (latitude, longitude, darkskyKey, callback) => {
+const forecast = (latitude, longitude, darkskyKey) => {
     const url = 'https://api.darksky.net/forecast/'+ darkskyKey + '/' + latitude + ',' + longitude + '?units=si'
-
-    request({url: url, json:true}, (error, {body}) => {
-        if (error) {
-            callback('Unable to connect to the service!', undefined)
-        } else if (body.error){
-            callback('Unable to find location, try another search', undefined)
-        }else {
-            callback(undefined, `Weather summary: ${body.currently.summary}. The highest temperature today is ${body.daily.data[0].temperatureHigh} degrees. The lowest temperature today is ${body.daily.data[0].temperatureLow} degrees. UX index is ${body.daily.data[0].uvIndex}. It is currently ${body.currently.temperature} degrees out. There is ${body.currently.precipProbability * 100}% chance of rain. `)
-        }
-        
+    return new Promise((resolve, reject) => {
```

Comment:
I used promise before, but not 100% clear about it yet. When should we use async/await and when do we use promise?They seem be to pretty similar to me! Is the promise chaining thing the best way to write promise as we did in the shopping cart one? Or in our case here it would be too difficult to read, and this way we write here is better?

Reply:
Async/await is actually just a fancy way to handle promises. I find it to be cleaner than promises for most things, but I still use promises all the time. Async await can only be used when you have a promise being returned to it. For example here are two ways to write the same code.

```
function doSomething() {
  return new Promise((resolve, reject) => ...)
}

// Promises
doSomething()
  .then(data => parseData(data))
  .catch(error => showError(error))

// Async
try {
  parseData(await doSomething())
} catch(error) {
  showError(error)
}
```

As you can see there isn't much difference and they are both pretty clean. The main time that async/await becomes cleaner is when you have lots of promise based code that depends on one another for example when using the mongoose library and querying the database.